sql/opt/exec/explain: TestMaximumMemoryUsage failed

sql/opt/exec/explain.TestMaximumMemoryUsage [failed](https://tanzanite.cluster.engflow.com/invocations/default/fbd166f3-9991-4db4-b726-0642c88566ef?testReportRun=4&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4cGxhaW46ZXhwbGFpbl90ZXN0) on release-24.3 @ [dd6ded593f157d28e104f7f5d7513145f099e625](https://github.com/cockroachdb/cockroach/commits/dd6ded593f157d28e104f7f5d7513145f099e625):


```
=== RUN   TestMaximumMemoryUsage
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestMaximumMemoryUsage4209898181
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    output_test.go:451: 
        	Error Trace:	pkg/sql/opt/exec/explain/output_test.go:451
        	Error:      	"0" is not greater than "5"
        	Test:       	TestMaximumMemoryUsage
        	Messages:   	expected maximum memory usage to be at least 5 MiB, full output:
        	            	
        	            	planning time: 399µs
        	            	execution time: 3.9s
        	            	distribution: full
        	            	vectorized: true
        	            	plan type: custom
        	            	rows decoded from KV: 10,000 (9.9 MiB, 10,000 gRPC calls)
        	            	cumulative time spent in KV: 4.8s
        	            	maximum memory usage: 0 B
        	            	network usage: 2.3 KiB (3 messages)
        	            	regions: test
        	            	sql cpu time: 29ms
        	            	estimated RUs consumed: 0
        	            	isolation level: serializable
        	            	priority: normal
        	            	quality of service: regular
        	            	
        	            	• group (hash)
        	            	│ group by: bucket
        	            	│
        	            	└── • scan
        	            	      sql nodes: n2, n3
        	            	      kv nodes: n2, n3
        	            	      regions: test
        	            	      actual row count: 10,000
        	            	      KV time: 4.8s
        	            	      KV contention time: 0µs
        	            	      KV rows decoded: 10,000
        	            	      KV bytes read: 9.9 MiB
        	            	      KV gRPC calls: 10,000
        	            	      estimated max memory allocated: 4.1 MiB
        	            	      sql cpu time: 20ms
        	            	      missing stats
        	            	      table: t@t_pkey
        	            	      spans: FULL SCAN
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestMaximumMemoryUsage4209898181
--- FAIL: TestMaximumMemoryUsage (10.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMaximumMemoryUsage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-50486

--- a/pkg/sql/distsql/server.go
+++ b/pkg/sql/distsql/server.go
@@ -354,6 +354,7 @@ func (ds *ServerImpl) setupFlow(
 		}
 		evalCtx.SetStmtTimestamp(timeutil.Unix(0 /* sec */, req.EvalContext.StmtTimestampNanos))
 		evalCtx.SetTxnTimestamp(timeutil.Unix(0 /* sec */, req.EvalContext.TxnTimestampNanos))
+		evalCtx.TestingKnobs.ForceProductionValues = req.EvalContext.TestingKnobsForceProductionValues
 	}
 
 	// Create the FlowCtx for the flow.

--- a/pkg/sql/execinfrapb/api.go
+++ b/pkg/sql/execinfrapb/api.go
@@ -48,9 +48,10 @@ func MakeEvalContext(evalCtx *eval.Context) EvalContext {
 	sessionDataProto := evalCtx.SessionData().SessionData
 	sessiondata.MarshalNonLocal(evalCtx.SessionData(), &sessionDataProto)
 	return EvalContext{
-		SessionData:        sessionDataProto,
-		StmtTimestampNanos: evalCtx.StmtTimestamp.UnixNano(),
-		TxnTimestampNanos:  evalCtx.TxnTimestamp.UnixNano(),
+		SessionData:                       sessionDataProto,
+		StmtTimestampNanos:                evalCtx.StmtTimestamp.UnixNano(),
+		TxnTimestampNanos:                 evalCtx.TxnTimestamp.UnixNano(),
+		TestingKnobsForceProductionValues: evalCtx.TestingKnobs.ForceProductionValues,
 	}
 }
 

--- a/pkg/sql/execinfrapb/api.proto
+++ b/pkg/sql/execinfrapb/api.proto
@@ -85,6 +85,7 @@ message EvalContext {
   optional int64 txn_timestamp_nanos = 2 [(gogoproto.nullable) = false];
   reserved 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14;
   optional sessiondatapb.SessionData session_data = 15 [(gogoproto.nullable) = false];
+  optional bool testing_knobs_force_production_values = 16  [(gogoproto.nullable) = false];
 }
 
 message SimpleResponse {

